### DISCUSSION TITLE
The "gaming OS" goal is confusing

### DISCUSSION THREAD

**Discussion by hardBSDk** [obhq/obliteration#1190] [Q&A] [answered]
AFAIK the README was updated to add the "gaming OS" goal with criticism to mainstream operating systems (I agree and disagree with some arguments).

I don't understand this goal because of the design of this emulator, it's adapted for PS4+ hardware and seems to rewrite the Orbis OS (FreeBSD) kernel in Rust.

What "metamorphosis" will happen?

**Comment by ultimaweapon** [comment, marked as answer]
> it's adapted for PS4+ hardware

I think this is the issue here. Our goal is not running on the PS4 hardware but to run the whole PS4 OS on other hardware like PC instead. Once this accomplished we will have a working kernel that already optimized for gaming since it is mimic the PS4 kernel. So our next step is to allows running non-PS4 binary like Windows or building our own SDK so people can port their game to run on our kernel.

The problem is English is my second language so sometime (or a lot) I made a mistake. It would be great if you or someone else can help us fixed this.

**Reply by hardBSDk** [reply]
There's no problem with the language, the explanation is good.

Have you researched the performance optimizations that Xbox (Windows) and Valve (Linux) did for their systems?

About Linux gaming performance, Wine and Valve allowed many Windows games to have equivalent or better performance than Windows.

I think you can import some ideas from Linux gaming, probably you already did.

**Reply by ultimaweapon** [reply]
Our goal is different from Wine. Wine provides a compatibility layer for Windows program to run by re-implementing the Windows DLLs but our goal is to provide system calls for those DLLs instead. That mean in order for the user to run Windows game on our kernel they need the whole system32 folder from the installed Windows. Of course Wine can also be ported to run on our kernel so people who don't have a Windows license can run Windows program but the compatibility may not good as using the system32 from Windows.

Linux is great but it have some problems that we already mentioned in the README we would like to solve. We want to see something like Steam Deck or Asus ROG Ally but running our kernel instead of Linux or Windows. One of the benefit of our kernel over Linux is the hardware vendor don't need to open-source any modification they made the same as Sony don't need to open-source PS4 kernel.

Xbox does not in our roadmap but we are happy to accept the contributions to support it.

**Reply by hardBSDk** [reply]
@ultimaweapon I think you misunderstood my question.

My question is: Do you imported the ideas behind the performance optimizations from Linux and Windows?

For example the FUTEX2 proposal from Proton and NTSYNC driver from Wine, they modify the Linux kernel to improve the Windows games performance.

I also asked if you imported performance optimizations from game consoles like Xbox, this is a complex topic because game consoles used a SoC design for a long time before PCs to extract maximum performance.

**Reply by ultimaweapon** [reply]
Indeed I was misunderstood your question.

> For example the FUTEX2 proposal from Proton and NTSYNC driver from Wine, they modify the Linux kernel to improve the Windows games performance.

It is too far for us to think about this right now. Our first goal is to mimic the PS4 kernel. Once this working we will explore what we can do to improve the performance.

> I also asked if you imported performance optimizations from game consoles like Xbox, this is a complex topic because game consoles used a SoC design for a long time before PCs to extract maximum performance.

AFAIK the only real different between the modern consoles and a PC is just a unified memory and how kernel manage time slice. Unlike legacy consoles that they have specialized hardware. For unified memory it might be possible with APU from AMD but may need supports from the UEFI firmware.
